### ISSUE TITLE
Good training and evaluation

### ISSUE THREAD

**Issue by yesencan** [CMM-25/a3#14]
Hi all,

I'm having some trouble doing 1.3-training. I tried changing the hyperparameters but failed to get a good result. During my most succesful run, I managed to make bob walk a little, though it is far from perfect. When I evaluated it, it completely fails to mimic to motion, and I could not find the reason why this happens.

https://github.com/user-attachments/assets/655a0291-4808-49d7-80d4-d98f8c038d46

https://github.com/user-attachments/assets/4086d7e1-480e-42a2-8e6d-a537cf6aba67

<img width="919" alt="Image" src="https://github.com/user-attachments/assets/685edd6d-9e65-4e80-b0f4-622119462bb3" />

Now, I'm wondering if my rewards and phase variable are implemented incorrectly or is the issue related to hyperparameters. Below are my rewards plot and phase plot:

![Image](https://github.com/user-attachments/assets/595d8876-2835-4de7-a27a-3dfe153dca5b)

![Image](https://github.com/user-attachments/assets/3b4f772a-843b-4423-b27e-49a43c913a62)

I'm a little scared to continue experimenting with the hyperparameters because I have already used 10 hours of server time, so I would greatly appreciate any advice on how to continue from here.

Best,
Yunus

**Comment by fzargarbashi** [comment]
Hii!

First some sanity check: are you specifying the correct model for evaluation by 'load_run' and 'checkpoint' arguments? If these are not specified, then by default the model from latest run will be used. The animation behaving differently than wandb video could indicate this is the cause.

Your plots look promising! Maybe something I would point from the plots is that the joint_target_rate reward is quite low (max about 0.2). Try to relax that a bit more to get higher rewards for that and simplify training.

Something that can't be seen in the plots is if you are updating the target_frame correctly or not. Double check that.

Another thing is make sure your training is converging, you can check that by mainly looking at the noise plot which should be decreasing and then mostly flat.

Best,
Fatemeh

**Comment by yesencan** [comment]
Thank you so much for the answer!

I double checked again to make sure I run the evaluation script correctly, and I do specify the load_run and checkpoint arguments. 

Regarding the joint_target_rate reward, I actually relaxed that quite a bit as you can see from the training plots (the first image). I may have relaxed it a bit too much (sigma = 7.5), however, making it stricter makes it so that the reward crashes to 0 and never go up.

I try to validate the target_frame update by removing it completely, which made the reference video to be stationary in the evaluation video. Running with the current implementation, it seems to be running the reference correctly. Do you think this is evidence that my implementation is correct? If not, do you have any advice on how to check its correctness?

The loss plots are below. I am not really sure how to interpret them, though I believe they mean the training converges?

<img width="938" alt="Image" src="https://github.com/user-attachments/assets/1fb0ea6d-a83f-4cbb-b846-5d6c6f56f819" />

**Comment by fzargarbashi** [comment]
The plots look reasonable, and it seems like you've done a good job trying to narrow things down.
However, because the evaluation videos are still not showing the expected behavior, it’s possible that there’s a subtle bug in the implementation that’s not reflected in the plots. If you look closely, even the wandb video doesn't walk like the reference motion (ignoring the global yaw which doesn't matter).

If you can’t figure it out by then, feel free to bring your code and results to the tutorial next week so we can look into it together more closely.

Best,
Fatemeh

**Comment by yesencan** [comment]
I will do that, thanks a lot!

**Comment by P10010** [comment]
@yesencan were you able to fix it? We are experiencing the exact same issue at the moment and have little idea on how to continue. We would really appreciate any hints

**Comment by yesencan** [comment]
@P10010 my implementation of the phase reset was incorrect. I'm not sure if you have the same issue but the reset function can be called for a subset of environments, while I reset all environments all the time, which was incorrect.
